src/main/java/view/createPostView.java

```diff
@@ -0,0 +1,209 @@
+                documentListenerHelper();  // Called for attribute changes (rarely needed for plain text input)
+            }
+        });
+    }
+
+    // Similarly, add for content field or other input fields
+    private void addPostContentListener() {
+        contentField.getDocument().addDocumentListener(new DocumentListener() {
+
+            private void documentListenerHelper() {
+                final CreatePostState currentState = createPostViewModel.getState();
+                currentState.setContent(contentField.getText());  // Update content in the state
+                createPostViewModel.setState(currentState);  // Set the updated state back
+            }
+
+            @Override
+            public void insertUpdate(DocumentEvent e) {
+                documentListenerHelper();  // Called when text is inserted
+            }
+
+            @Override
+            public void removeUpdate(DocumentEvent e) {
+                documentListenerHelper();  // Called when text is removed
+            }
+
+            @Override
+            public void changedUpdate(DocumentEvent e) {
+                documentListenerHelper();  // Called for attribute changes (rarely needed for plain text input)
+            }
+        });
+    }
+
+
+    private void addSaveButtonListener() {
+        saveButton.addActionListener(e -> {
+            // Gather input data from the user interface
+            String author = homepageController.fetchUser().getUsername(); // TODO: username not name right?
+            String content = contentField.getText();
+            String attachmentPath = null; // TODO: What do we use here
+            String fileType = "text"; // Assume fileType is determined elsewhere
+            int dislikes = 0; // Default value
+            int likes = 0; // Default value
+            String postTitle = titleField.getText();
+            List moderators = new List(); // TODO: Are these two correct
+            List comments = new List();
+            String category = (String) categoryDropdown.getSelectedItem();
+
+            // Validate required inputs
+            // TODO: Not sure how to test if category is null is it categoryDropdown.getSelectedItem() = null?
+            if (postTitle.isEmpty() || content.isEmpty() || author.isEmpty() ) {
+                JOptionPane.showMessageDialog(this, "Title, content, and author are required!");
+                return;
+            }
+
+            // Pass the input data to the controller
+            createPostController.execute(homepageController.fetchUser().getUserID(), content, attachmentPath, fileType, dislikes, likes, postTitle, (java.util.List<User>) moderators, (java.util.List<Comment>) comments, category);
+            // Provide user feedback and switch to the homepage
+            JOptionPane.showMessageDialog(this, "Post created successfully!");
+            createPostController.switchToHomePageview();
+        });
```

Comment:
I think after saving you need to explicitly close the dialog no?

```suggestion
this.setVisible(false)
            createPostController.switchToHomePageview();
        });
```